tensorflow_lite_support/cc/task/vision/landmark_detector.cc

```diff
@@ -0,0 +1,157 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+    
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow_lite_support/cc/task/vision/landmark_detector.h"
+
+#include "absl/algorithm/container.h"
+#include "absl/strings/str_format.h"
+#include "absl/strings/string_view.h"
+#include "flatbuffers/flatbuffers.h"
+#include "tensorflow_lite_support/cc/common.h"
+#include "tensorflow_lite_support/cc/port/integral_types.h"
+#include "tensorflow_lite_support/cc/port/status_macros.h"
+#include "tensorflow_lite_support/cc/task/core/task_api_factory.h"
+#include "tensorflow_lite_support/cc/task/core/task_utils.h"
+#include "tensorflow_lite_support/cc/task/core/tflite_engine.h"
+#include "tensorflow_lite_support/cc/task/vision/proto/landmarks_proto_inc.h"
+#include "tensorflow_lite_support/cc/task/vision/utils/frame_buffer_utils.h"
+#include "tensorflow_lite_support/metadata/cc/metadata_extractor.h"
+#include "tensorflow_lite_support/metadata/metadata_schema_generated.h"
+
+namespace tflite {
+namespace task {
+namespace vision {
+
+namespace {
+
+using ::absl::StatusCode;
+using ::tflite::metadata::ModelMetadataExtractor;
+using ::tflite::support::CreateStatusWithPayload;
+using ::tflite::support::StatusOr;
+using ::tflite::support::TfLiteSupportStatus;
+using ::tflite::task::core::AssertAndReturnTypedTensor;
+using ::tflite::task::core::TaskAPIFactory;
+using ::tflite::task::core::TfLiteEngine;
+
+}  // namespace
+
+/* static */
+StatusOr<std::unique_ptr<LandmarkDetector>> LandmarkDetector::CreateFromOptions(
+    const LandmarkDetectorOptions& options) {
+  RETURN_IF_ERROR(SanityCheckOptions(options));
+
+  // Copy options to ensure the ExternalFile outlives the constructed object.
+  auto options_copy = absl::make_unique<LandmarkDetectorOptions>(options);
+
+  std::unique_ptr<LandmarkDetector> landmark_detector;
+  if (options_copy->base_options().has_model_file()) {
+    ASSIGN_OR_RETURN(landmark_detector,
+                     TaskAPIFactory::CreateFromBaseOptions<LandmarkDetector>(
+                         &options_copy->base_options()));
+  } else {
+    // Should never happen because of SanityCheckOptions.
+    return CreateStatusWithPayload(
+        StatusCode::kInvalidArgument,
+        absl::StrFormat("Missing mandatory `model_file` field in `base_options`"),
+        TfLiteSupportStatus::kInvalidArgumentError);
+  }

```

Comment:
Replace this with
```
ASSIGN_OR_RETURN(auto landmark_detector, 
                                              TaskAPIFactory::CreateFromBaseOptions<LandmarkDetector>(
                                                      &options_copy->base_options()));
```
?

Reply:
I am getting errors if I use auto or don't use else statement.

Reply:
Yeah, I remember you mentioned that. What is the error message you get?

Reply:
IDK, it is not showing anything specific

Reply:
Can you please the log here, so that I can take a look?

Reply:
exec ${PAGER:-/usr/bin/less} "$0" || exit 1
Executing tests from //tensorflow_lite_support/cc/test/task/vision:landmark_detector_test
-----------------------------------------------------------------------------
Running main() from gmock_main.cc
[==========] Running 2 tests from 2 test suites.
[----------] Global test environment set-up.
[----------] 1 test from CreateFromOptionsTest
[ RUN      ] CreateFromOptionsTest.FailsWithMissingModel
 test.log

Reply:
Ha, I forgot to mention, you should add `--test_output=all` to your bazel command, and then you can see the full message.

Reply:
!bazel test --test_output=all //tensorflow_lite_support/cc/test/task/vision:all
I am already using it

Reply:
Alright. I'll help you debug this next week.

Reply:
Ok